### PR TITLE
Added Panthera

### PR DESCRIPTION
**Pull request by Eogos**
| Q | A |
| --- | --- |
| Bug fix? | No |
| New feature? | Yes |
| Needs wipe? | No |
| Fixed issues | N/A |

### Description:
Adds a Liberation mission.sqm for the map Panthera

### Content:
- [x] Panthera mission.sqm

### Successfully tested on:
- [x] Local MP
- [x] Dedicated MP

### CODE REVIEW

**Comment by Wyqer** [comment]
Thanks for your contribution.

I'll have a look at the mission.sqm during the weekend, as I've not enough time to do that in enough detail tomorrow.

In general I would, after possible needed adjustments, add it as a basefile, but I wouldn't publish it on the workshop by myself, as that would need me to maintain it additionally to all the other maps we've published. So the general idea would be that we merge it (if it gets a positive review) and with release of 0.96.5, I'll hand you over a pbo with the mission, which you can publish and maintain on the workshop yourself.

At least that's my current thought. We could also have some talk about this during the weekend in the contributors channel on Discord.

**Comment by Eogos** [comment]
> So the general idea would be that we merge it (if it gets a positive review) and with release of 0.96.5, I'll hand you over a pbo with the mission, which you can publish and maintain on the workshop yourself.

That should be fine, though updates to the new framework may be slow/late from time to time.

**Comment by Wyqer** [comment]
Had a brief look at the mission.sqm
* Remove the Liberty and use the Freedom like in the other legacy framework maps
* Remove all unnecessary "immersive" clutter stuff at the placed bases/locations for performance reasons. Have a look at the other official maps with placed bases. It's basically buildings/walls and nothing else. No crates, garbage, furniture, etc.
* Make sure it has the same attributes like all the other mission files concerning strings, eden settings, etc.

**Comment by Eogos** [comment]
> Remove all unnecessary "immersive" clutter stuff at the placed bases/locations for performance reasons. Have a look at the other official maps with placed bases. It's basically buildings/walls and nothing else. No crates, garbage, furniture, etc.

Damn, alright will do when I can. The rest should be fairly straight forward to fix as well.

**Comment by Eogos** [comment]
@Wyqer Actually, would it be ok to switch them to simple objects for some of them instead of outright removing all of them? I would like to keep some of them in like the helipad areas and what not so it isn't just an open square.

**Comment by Wyqer** [comment]
It's our convention to keep it as simple as possible, so removal would be the way to go. Best you orientate on Altis or Chernarus for a feeling of what could stay and how "empty" is should be.

I can understand that you want to keep it "nice and immersive", I've done the same at the beginning.
But the players wouldn't really notice it at an enemy base. Especially they wouldn't notice "missing immersive clutter stuff" in the end, when fighting against tanks and infantry in the area.

**Comment by Eogos** [comment]
> There are still many grass cutters / hide terrain modules and I don't know if they're really necessary.

Most of those are from one specific sector on the North West island. There is an abandoned factory sitting out in the middle of a field. I turned it into a proper factory complex but had to remove the grass and wheat/trees/rocks. The grass cutters are pretty self explanatory but the wheat, trees, and rocks had to be removed by hide terrain modules.

TL;DR Yes they're necessary unless you want me to remove an entire factory sector _which please don't it took me like a solid hour to get that complex into a way that looked nice_

> Medic buildings not declared as ACE medical facility.
I assume that's as simple as dropping a line of code in the init of the building on the Freedom?

**Comment by Eogos** [comment]
For context, here is some pictures of the complex I made:
https://i.imgur.com/YFR2msS.jpg
https://i.imgur.com/KHHuEkK.jpg

rather proud of it tbh, it looks quite nice imo. AI even respects the fencing around it and goes through the gate which I wasn't expecting when I built it if I'm honest.

**Comment by Dubjunk** [comment]
Yeah the factory looks good but there's quite much stuff around it.
I can't assess if it'll affect the performance.

**Comment by Eogos** [comment]
> 
> 
> Yeah the factory looks good but there's quite much stuff around it.
> I can't assess if it'll affect the performance.

It really shouldn't. If someones already having bad performance in that area removing those objects won't do a whole lot. The only performance issues people might have is number of AI but I've already addressed a lot of that from the original version by removing sectors in places that were spawning too many AI before the limiter could kick in.

I know it's not meaning as much from me cause I built it so of course it runs fine for me but it's 9 buildings (not counting ones there by default), some lights, fencing, and runway texture objects. Other than that everything in there is hide terrain modules and grass cutters which if anything help performance rather than hinder it.

**Comment by Wyqer** [comment]
Looks fine so far.
You just missing the ACE medical facility attributes for the medical buildings in your placed base `Medical Complex` layer.
```sqf
this setVariable ["ace_medical_isMedicalFacility", true, true];
```
needs to be in their init field.

Are you also sure that your ignore list is complete? Or asked in another way: Can you really ensure you've catched anything which should be ignored?

After that it should be fine and would get my approval. I guess I can also publish and maintain it directly on the workshop to avoid mission file sending to you, so that you update, etc.

**Comment by Eogos** [comment]
> You just missing the ACE medical facility attributes for the medical buildings in your placed base `Medical Complex` layer.
> 
> ```sqf
> this setVariable ["ace_medical_isMedicalFacility", true, true];
> ```
> 
> needs to be in their init field.

Can do, will make another PR tonight with that fixed.



> Are you also sure that your ignore list is complete? Or asked in another way: Can you really ensure you've catched anything which should be ignored?

Nope. When I initially made this for my group I was unaware of how the ignore lists worked and didn't make one, for this I copied the Sahrani one because they have similar building and vegetation usage for the most part at the suggestion of @veteran29 

It _should_ catch a lot of them as Panthera does use CUP buildings, but there may be some missing.

**Comment by Wyqer** [comment]
> It should catch a lot of them as Panthera does use CUP buildings, but there may be some missing.

Please recheck it once and make sure it's covering everything basically. 👍

**Comment by Eogos** [comment]
> Please recheck it once and make sure it's covering everything basically.

Everything should be caught now as of KPLP5
